### DISCUSSION TITLE
What is an API with .NET?  - Page 289

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#113] [Q&A] [answered]
Hi,

You wrote about "APIs with .NET" but you did not give any explanation about "API".

In my opinion, the book needs a complete explanation of  "What is an API?" and "What are "APIs with .NET"?"?

Page 289 - Why Asynchronous Programming Is Important

**Comment by christiannagel** [comment, marked as answer]
I'm converting this issue to a discussion

**Reply by christiannagel** [reply]
Sorry this was not clear. An API is "Application Programming Interface". All the methods of .NET classes are the APIs of .NET.

Maybe interesting with this aspect, the .NET Standard documentation shows the number of APIs available with each version:

[Select .NET Standard version](https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-1-0#select-net-standard-version)

Maybe this was not clear, and I should have explained it differently (page 289):
"Many APIs with .NET offer both a synchronous and an asynchronous version." This is not really what I meant. This should be like this:
"Many APIs with .NET that offer asynchronous versions also offer a synchronous version".

Does this answer the question?

**Reply by ShervanN** [reply]
Yes, Thank you.
